refsheet.cls

```diff
@@ -1,57 +1,212 @@
+%
+\setcounter{secnumdepth}{0}
+%
+%   No indentation and superfluous spacing
+%
+\setlength{\parindent}{0pt} 
+\titlespacing{\section}{0pt}{0pt}{0pt}
+\titlespacing{\subsection}{0pt}{0pt}{0pt}
+%
+%%%
+
+%%% COMMANDS AND ENVIRONMENTS
+%
+%   Extended definition of lists with enumitem
+%
+\RequirePackage[inline]{enumitem}[2011/08/28]
+%   Calculating widths, e.g. with \widthof
+\RequirePackage{calc} 
+
+% generate dense lists
+\setlist{noitemsep,topsep=0.05ex,}
+\setlist[itemize]{label=\textendash,leftmargin=*}
+\setlist[description]{leftmargin=!}
+% Provide a list-type for truetype descriptions
+% This is retained for backward compatibility
+\newlist{ttdesc}{description}{1}
+\setlist[ttdesc]{font=\ttfamily}
+% Conveniently assigning the longest label as 
+\newenvironment{rslisttt}[1]
+{
+  \begin{description}[labelwidth=\widthof{\texttt{#1}},font=\ttfamily]
+}{
+  \end{description}
+}
 
-%% Title area
+% Provide a list-type for normal descriptions
+\newenvironment{rslist}[1]
+{
+    \begin{description}[labelwidth=\widthof{#1}]
+}{
+    \end{description}
+}
 
-% \RequirePackage{titling} % Unnecessary
-\renewcommand{\maketitle}{%
-  {\begin{center}\Large \@title \\ \vspace{4pt} \small \@author, \@date\end{center}}%
+% Provide a list-type for bold descriptions
+\newenvironment{rslistbf}[1]
+{
+    \begin{description}[labelwidth=\widthof{\textbf{#1}},font=\bfseries]
+}{
+    \end{description}
 }
 
+% Provide a list-type for italic descriptions
+\newenvironment{rslistit}[1]
+{
+    \begin{description}[labelwidth=\widthof{\textit{#1}},font=\itshape]
+}{
+    \end{description}
+}
 
-%% Document Divisions
+% Provide inline description lists
+\newenvironment{rsinline}[1][\ttfamily]
+{
+  \begin{description*}[font={#1}]
+}{
+    \end{description*}
+}
 
-\RequirePackage{titlesec}
-\RequirePackage{nopageno} % To keep \section from resetting page style
+% Provide column list
+\newenvironment{rscolslist}[1][2]
+{
+  \begin{multicols*}{#1}
+    \begin{itemize*}[itemjoin={{\newline}},afterlabel={\nobreakspace}]
+}{
+    \end{itemize*}
+  \end{multicols*}
+}
 
-\setcounter{secnumdepth}{0} % no section numbering
-\setlength{\parindent}{0pt} % disabling indentation by default
-\titlespacing{\section}{0pt}{0pt}{0pt}
-\titlespacing{\subsection}{0pt}{0pt}{0pt}
 
+% Provide an easier interface to create a table
+\RequirePackage{tabularx}
 
-%% Lists
+% Full width table with automagic column widths
+\newcounter{rstablecols}
+\newenvironment{rstable}[2][X]
+{
+  \setcounter{rstablecols}{#2}
+  \addtocounter{rstablecols}{-1}
 
-\RequirePackage{enumitem} % for consistent formatting of lists
-\RequirePackage{calc} % for \widthof
+  \tabularx{\linewidth}{ *{\value{rstablecols}}{#1} X }%
+}{
+  \endtabularx
+  \setcounter{rstablecols}{0}
+}
 
-% Provide a list-type for commands and descriptions
-\newlist{ttdesc}{description}{1}
-\setlist[ttdesc]{font=\ttfamily,noitemsep}
+\newenvironment{rsmathtable}
+{
+  \tabularx{\linewidth}{ >{\(\displaystyle}l<{\)} X }%
```

Comment:
Please use `$` notation instead of `\(` and `\)`. See [this TeX.SX question](https://tex.stackexchange.com/q/510/17423) (particularly [Frank's comment](https://tex.stackexchange.com/questions/510/are-and-preferable-to-dollar-signs-for-math-mode/59195#comment83898_513)).